plugins/database/src/db_plugin.cpp

```diff
@@ -2141,23 +2163,26 @@ irods::error db_mod_data_obj_meta_op(
         rodsLog( LOG_SQL, "chlModDataObjMeta" );
     }
 
-    adminMode = 0;
-    theVal = getValByKey( _reg_param, ADMIN_KW );
-    if ( theVal != NULL ) {
-        adminMode = 1;
+    bool adminMode{};
+    if (getValByKey(_reg_param, ADMIN_KW)) {
+        if (LOCAL_PRIV_USER_AUTH != _ctx.comm()->clientUser.authInfo.authFlag) {
+            return ERROR(CAT_INSUFFICIENT_PRIVILEGE_LEVEL, "failed with insufficient privilege");
+        }
+        adminMode = true;
     }
 
     std::string update_resc_id_str;
 
     bool update_resc_id = false;
     /* Set up the updateCols and updateVals arrays */
-    for ( i = 0, j = 0; strcmp( regParamNames[i], "END" ); i++ ) {
-        theVal = getValByKey( _reg_param, regParamNames[i] );
-        if ( theVal != NULL ) {
-            if( std::string( "resc_name") == colNames[i]) {
+    std::size_t i = 0, j = 0;
+    for (i = 0, j = 0; i < regParamNames.size(); i++) {
+        theVal = getValByKey(_reg_param, regParamNames[i].data());
+        if (theVal) {
+            if(std::string( "resc_name") == colNames[i]) {
                 continue;
             }
-            else if( std::string( "resc_hier") == colNames[i]) {
+            else if(std::string( "resc_hier") == colNames[i]) {
```

Comment:
the `std::string` wrapping `"resc_hier"` is probably not needed given `colNames[i]` is a `std::string_view`.

Reply:
Good catch - will fix